### ISSUE TITLE
Helpers for identity's data

### ISSUE THREAD

**Issue by JohnnyMcWeed** [luyadev/luya-privacy#6]
I think on the long term this extension has to include some helper functionality for an identity's privacy.

What I mean is, that there should be "a default way to handle an identity's privacy".
So there should helpers be inlcuded, something like:
- getDataByIdentity($identity)
- deleteDataByIdentity($identity)

Then new modules could easily implement these methods and e. g. a user component can go through all modules, collect everything together and display it to the user. Moreover it make's the delete task very easy.

**Comment by nadar** [comment]
What would `$identity` look like?

**Comment by JohnnyMcWeed** [comment]
It would be the user identity... Maybe user is the better word.

E. g. 
Implemented in some module, so that it returns all data which belongs to that user:
```
getDataByUser($user_id) {
    ...
    return $userData
}
``` 
User component could then grab these data for all modules and display/export it to the user
```
getUserData($user_id) {
    ...
    foreach($modules as $module) {
        $data .= $module->getDataByUser($user_id);
    }
    ...
    return $data;
}
```
 And the same for delete. That's the idea.

I think like this using other modules works like a charm in terms of end users who want to control their data.

**Comment by nadar** [comment]
currently i don't see any need for such a integration. You are controlling your application so you know where the data is stored and it would be hard to unify this information. What you could use to collect users is the implementation of an Interface into the specific class and then generate a module where you have to specific the classes which integrates those interfaces ... but currently i think this is a huge overkill, but maybe in future.

**Comment by JohnnyMcWeed** [comment]
I agree that it is an overkill right now, but I think an standard interface would be great and become very handy on long term. 
Sure I am controlling my application by myself and yes I know where my data is stored. But if you decide to fasten up the development process by using other people's modules, you have to implement a lot more. Moreover different people have to do that same task again.
On the other hand, with an interface the chance is bigger that people implement against it. It has only to be done once, and it would be more "standardized", which makes it easier to use.

**Comment by nadar** [comment]
@JohnnyMcWeed so we would have to make a concept of what the interface could look like. Any suggestions? Because then we could at least provide the interface, and think about how to call and evaluate those later.
